internal/provider/cluster_resource.go

```diff
@@ -315,6 +316,10 @@ func (r *clusterResource) Create(
 		if plan.Regions != nil {
 			regionNodes := make(map[string]int32, len(plan.Regions))
 			for _, region := range plan.Regions {
+				if region.Primary.ValueBool() {
```

Comment:
You can actually do this validation at plan time in `ConfigValidators` instead. I think something like this should work:
```
resourcevalidator.Conflicting(
	path.MatchRoot("regions").AtAnyListIndex().AtName("primary"),
	path.MatchRoot("dedicated"),
),
```

Reply:
Done!